docs/src/references/agent/software-management.md

```diff
@@ -0,0 +1,332 @@
+---
+title: Software management
+tags: [Reference, Agent, Software Management]
+sidebar_position: 3
+---
+
+Thin-edge software management is implemented by the `software-update` operation,
+which gives the ability to:
+- manage software packages of different types on the same device 
+- fetch a pertinent subset of the software packages installed on a device
+- install, update and remove software packages on a device
+
+## MQTT API
+
+The `software-update` operation API follows the [generic thin-edge rules for operations](device-management-api.md):
+- The `te/<device-topic-id>/cmd/software-update` topic is used to publish the type of software packages
+  that can be managed on the device with the given topic identifier.
+- Each `te/<device-topic-id>/cmd/software-update/+` topic is dedicated to a software update command instance,
+  where all the subsequent states of this instance are published during its execution.
+- The workflow is [generic with `"init"`, `"executing"`, `"successful"` and `"failed"` statutes](references/agent/device-management-api.md/#operation-workflow).
+
+### Operation registration
+
+The registration message of the `software-update` operation on a device:
+- must provide a `supportedTypes` list of the types of software package that can be installed on this device (e.g. `["apt", "docker"]`)
+- can provide a description of the operation and of each supported package type.
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update' '{
+    "description": "Install, update and remove software packages",
+    "supportedTypes": [
+      "apt",
+      "docker",
```

Comment:
Removing trailing comma as they are not supported in the vanilla JSON spec

```suggestion
      "docker"
```

Reply:
fixed